geps/gep-3792/index.md

```diff
@@ -0,0 +1,190 @@
+# GEP-3792: External Gateways
+
+* Issue: [#3792](https://github.com/kubernetes-sigs/gateway-api/issues/3792)
+* Status: Provisional
+
+(See [status definitions](../overview.md#gep-states).)
+
+## User Story
+
+**[Chihiro] and [Ian] want a way for out-of-cluster Gateways to be able to
+usefully participate in a GAMMA-compliant in-cluster service mesh.**
+
+Historically, API gateways and ingress controllers have often been implemented
+using a Service of type LoadBalancer fronting a Kubernetes pod running a
+proxy. This is simple to reason about, easy to manage for sidecar meshes, and
+will presumably be an important implementation mechanism for the foreseeable
+future. Some cloud providers, though, are moving the proxy outside of the
+cluster, for various reasons which are out of the scope of this GEP. Chihiro
```

Comment:
Could you elaborate on one or two of those reasons?  It would help to have use cases.

Reply:
I'll defer this one to @robscott, @keithmattix, and @mikemorris, who are all working with cloud providers who're looking at doing this. I may or may not be able to correctly summarize their reasons.

I personally tend to believe that it is easier to reason about Kubernetes Gateways if, y'know, they run _in Kubernetes_. Shockingly, I have been unable to convince major cloud providers that they should change their plans due solely to my opinion. 😇

Reply:
A primary reason is that off-cluster gateways can be used to offer a cleaner "managed" experience, where a "logical" gateway can scale much easier without requiring users to be responsible for managing underlying compute/memory resources, potentially be consolidated onto purpose-built infrastructure hardware or networking, offer alternative billing models (like per request instead of trying to estimate needed compute costs), integrate more seamlessly into other cloud product offerings, etc